src/main/java/io/numaproj/numaflow/sourcer/ReadRequest.java

```diff
@@ -0,0 +1,18 @@
+package io.numaproj.numaflow.sourcer;
+
+import java.time.Instant;
+
+/**
+ * ReadRequest request for reading messages from source.
+ */
+public interface ReadRequest {
+    /**
+     * @return the number of messages to be read
+     */
+    long getCount();
+
+    /**
+     * @return the timeout for reading messages
+     */
+    Instant getTimeout();
```

Comment:
Is Instant a specific time instead of a duration?

Reply:
Duration is better for this case, updated it.